Xamarin.Forms.Platform.Android/Extensions/BrushExtensions.cs

```diff
@@ -0,0 +1,236 @@
+﻿using System;
+using Android.Graphics;
+using Android.Graphics.Drawables;
+using Android.Graphics.Drawables.Shapes;
+using static Android.Graphics.Drawables.GradientDrawable;
+using AView = Android.Views.View;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	public static class BrushExtensions
+	{
+		public static void UpdateBackground(this AView view, Brush brush)
+		{
+			if (view == null)
+				return;
+
+			if (view.Background is GradientStrokeDrawable)
+			{
+				// Remove previous background gradient if any
+				view.SetBackground(null);
+			}
+
+			if (brush == null || brush.IsEmpty)
+				return;
+
+			if (brush is LinearGradientBrush linearGradientBrush)
+			{
+				GradientStopCollection gradients = linearGradientBrush.GradientStops;
+
+				if (!IsValidGradient(gradients))
+					return;
+			}
+
+			if (brush is RadialGradientBrush radialGradientBrush)
+			{
+				GradientStopCollection gradients = radialGradientBrush.GradientStops;
+
+				if (!IsValidGradient(gradients))
+					return;
+			}
+
+			view.SetPaintGradient(brush);
+		}
+
+		public static void UpdateBackground(this Paint paint, Brush brush, int height, int width)
+		{
+			if (paint == null || brush == null || brush.IsEmpty)
+				return;
+
+			if (brush is SolidColorBrush solidColorBrush)
```

Comment:
We don't have to do this now but is there a way we can generalize this out? Instead of using if statements? Maybe we need to create a brush handler like we have with ImageSource handler?

Reply:
We could create handlers in the same way as we have with the images.

Reply:
Yea, maybe create a new issue/spec to cover this down the road?